### DISCUSSION TITLE
overlap between nvim statusline (and associated plugins) and built-in statusline manager

### DISCUSSION THREAD

**Discussion by mkaminsky** [vscode-neovim/vscode-neovim#1935] [Q&A]
I've been playing around with the new nvim statusline feature in v1.10.0 (PR #1895), which is really nice.

Specifically, I've been trying to see if I can create a modified version of the [lualine.nvim](https://github.com/nvim-lualine/lualine.nvim) configuration that I use outside of vscode.  For example, I am able to load it successfully, via lazy.nvim, as follows:
```lua
  {
    "nvim-lualine/lualine.nvim",
    vscode = true,
    opts = function()
      return {
        options = {
          section_separators = "",
          component_separators = "|",
        },
        sections = {
          lualine_a = { "mode", "selectioncount" },
          lualine_b = {},
          lualine_c = {},
          lualine_x = {},
          lualine_y = {},
          lualine_z = {},
        },
      }
    end,
  }
```

In vscode, the statusline will show duplicate information in, for example, visual mode:     `VISUAL | 4|4`
  - The first 4 comes from lualine's "selectioncount"; the second is from the extension.
  - It seems that the statusline is made up a handful of components:
    `this.statusBar.text = [this.nvimStatusLine, this._modeText, this._cmdText, this._msgText]`
  -  `cmdText` is showing the same selection information as the `nvimStatusLine`
  - But...`cmdText` also shows partial commands, such as a "r" and the count prefix, which the lualine config does not.
 - Is there a way to have the extension show just the command and not the selection part, or to have it show nothing, and then somehow get lualine/nvim to show both (which perhaps is more of a lualine/nvim question, but I wasn't yet able to get that to work.  I thought "%S" would do it, but doesn't seem to work).

Also, I would have thought that `modeText` should show "VISUAL" and the mode would appear twice, but it doesn't.

More generally, is it possible to show all of the information provided by `modeText`, `cmdText`, and `msgText` directly via the statusline variable (or a plugin), such that vscode-neovim doesn't need to do so directly?

Thanks to everyone for the great extension!

**Comment by xiyaowong** [comment]
`:h 'showmode'`
`:h 'shortmess'`
